js/web/lib/wasm/session-handler.ts

```diff
@@ -16,35 +17,71 @@ export class OnnxruntimeWebAssemblySessionHandler implements SessionHandler {
   inputNames: string[];
   outputNames: string[];
 
-  async createSessionAllocate(path: string): Promise<SerializableModeldata> {
-    // fetch model from url and move to wasm heap. The arraybufffer that held the http
-    // response is freed once we return
-    const response = await fetch(path);
-    const arrayBuffer = await response.arrayBuffer();
-    return createSessionAllocate(new Uint8Array(arrayBuffer));
+  async fetchModelAndWeights(modelPath: string, weightsPath?: string): Promise<[Uint8Array, ArrayBuffer?]> {
+    const modelResponse = await fetch(modelPath);
+    const promises: [Promise<Uint8Array>, Promise<ArrayBuffer>?] = [
+      modelResponse.arrayBuffer().then(b => new Uint8Array(b))
+    ];
+
+    if (weightsPath) {
+      const weightsResponse = await fetch(weightsPath);
+      const weightsSize = parseInt(weightsResponse.headers.get('Content-Length')!, 10);
+      // we cannot create ArrayBuffer > 2gb but 64bit WASM Memory can have arbitrary size
+      const weightsMemory = new WebAssembly.Memory({
```

Comment:
what is the advantage for using `new WebAssembly.Memory()` than using `new ArrayBuffer(weightSize)`?

Reply:
If weightSize is > 2gb then `new ArrayBuffer` will fail with range error. Wasm memory can allocate any size (under global restriction of 16gb per browser tab)